src/cli/cli_tcat.cpp

```diff
@@ -109,6 +113,84 @@ static void HandleBleSecureReceive(otInstance               *aInstance,
     IgnoreReturnValue(otBleSecureFlush(aInstance));
 }
 
+template <> otError Tcat::Process<Cmd("vendorid")>(Arg aArgs[])
+{
+    otError                  error = OT_ERROR_NONE;
+    otTcatDeviceId           devId;
+    static const char *const kVendorIdTypes[] = {"empty", "oui24", "oui36", "discriminator", "ianapen"};
+
+    mVendorInfo.mDeviceIds = sVendorDeviceIds;
+
+    if (aArgs[0].IsEmpty())
+    {
+        if (mVendorInfo.mDeviceIds[0].mDeviceIdType != OT_TCAT_DEVICE_ID_EMPTY)
+        {
+            OutputLine("Set vendorIds:");
+            for (size_t i = 0; mVendorInfo.mDeviceIds[i].mDeviceIdType != OT_TCAT_DEVICE_ID_EMPTY; i++)
+            {
+                OutputFormat("type %s, value: ", kVendorIdTypes[mVendorInfo.mDeviceIds[i].mDeviceIdType]);
+                OutputBytesLine((uint8_t *)mVendorInfo.mDeviceIds[i].mDeviceId, mVendorInfo.mDeviceIds[i].mDeviceIdLen);
+            }
+        }
+        else
+        {
+            OutputLine("%s", kVendorIdTypes[OT_TCAT_DEVICE_ID_EMPTY]);
+        }
+        ExitNow();
+    }
+
+    if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_OUI24])
+    {
+        devId.mDeviceIdType = OT_TCAT_DEVICE_ID_OUI24;
+    }
+    else if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_OUI36])
+    {
+        devId.mDeviceIdType = OT_TCAT_DEVICE_ID_OUI36;
+    }
+    else if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_DISCRIMINATOR])
+    {
+        devId.mDeviceIdType = OT_TCAT_DEVICE_ID_DISCRIMINATOR;
+    }
+    else if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_IANAPEN])
+    {
+        devId.mDeviceIdType = OT_TCAT_DEVICE_ID_IANAPEN;
+    }
+    else if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_EMPTY])
+    {
+        for (auto &vendorDeviceId : sVendorDeviceIds)
```

Comment:
Avoid the use `auto`? 
We recommend using explicit types in `for` loop (make it easier to read the code later).

Reply:
fixed.